src/shared/sleep-config.c

```diff
@@ -134,6 +134,51 @@ int read_battery_capacity(void) {
         return battery_capacity;
 }
 
+int get_discharge_rate(void) {
+        _cleanup_free_ char *dis_rate;
+        int stored_discharge_rate;
+        const char *dirpath;
+        sd_id128_t *machine_id = NULL;
+        int r;
+
+        r = sd_id128_get_machine(machine_id);
+        if(r < 0)
+                return r;
+
+        dirpath = path_join("/var/lib/systemd/sleep", machine_id, "/battery_discharge_rate");
+
+        r = read_one_line_file(dirpath, &dis_rate);
+        if (r < 0 || r == -ENOENT)
+               return r;
+
+        r = safe_atoi(dis_rate, &stored_discharge_rate);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to convert discharge rate from char value to int: %m");
```

Comment:
mention the file path too in the error message, otherwise it's confusing

Reply:
this error is for failure to convert discharge rate. Discharge rate is fecthed before this. Added now in above code when read_one_line_file is called

Reply:
yes but it is useful to tell exactly where the unreadable value is coming from, otherwise from just this message a user has no way of knowing